checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleAuditListener.java

```diff
@@ -100,10 +100,11 @@ public void addError(AuditEvent event) {
             final NewIssue issue = context.newIssue();
             final ActiveRule rule = ruleFinder.find(
                     RuleKey.of(CheckstyleConstants.REPOSITORY_KEY, ruleKey));
+            final Integer lineId = getLineId(event);
             if (Objects.nonNull(issue) && Objects.nonNull(rule)) {
                 final NewIssueLocation location = issue.newLocation()
                         .on(currentResource)
-                        .at(currentResource.selectLine(getLineId(event)))
+                        .at(currentResource.selectLine(lineId != null ? lineId.intValue() : 1))
```

Comment:
`getLineId` is only used in this class and method by us besides tests. Why do we want the method to keep returning `null` when no one will use the `null` value. Shouldn't we just change it's default to `1` and have it return `int` instead?

Reply:
True, I chose the least invasive change that fixes the problem. After it's fixed, I agree that it may be a good idea to refactor `getLineId()` as well.  
But I must admit that I don't know enough about the underlying mechanics of SonarQube - maybe they really do use `null` to signify that an issue relates to the entire file. Then by changing these semantics for us, we might break conceptual integrity. I didn't want to take any risks here. 😉

Reply:
> maybe they really do use null to signify that an issue relates to the entire file

The interface for InputFile says it can't be `null` by its definition,
> `TextRange selectLine(int line);`

This is what is causing the auto-unboxing. If it is true they allow `null` the only way to distinguish this is to call `NewIssueLocation#at` with the entire range of the file. As you mentioned this is related to https://github.com/checkstyle/checkstyle/issues/6000 , which stated it was better to print violation on some physical line in file, hence why we picked line 1. So we are no longer printing violations on entire file.

So I see no issues making the change.

Reply:
(Well, line 1 might be problematic if the file is empty. This will have to be handled in IDE plugins, which must deal with the fact that issues are reported on non-existent lines. Not a big deal, but as we see, it should be handled consistently throughout our little Checkstyle ecosystem. But that's really a Checkstyle issue rather than one here.)
Do you need *me* to also do the refactoring of `getLineId()`? Or would it be acceptable to have this fix first, and do the refactoring later?

Reply:
> Well, line 1 might be problematic if the file is empty.

Then this is also an issue with your change, since you are doing the same thing I am asking for just in a different place. You are replacing `null` with `1` without checking the size of the file. We changed all 0s to 1s and didn't do any change here to check number of lines in file being reported, so we don't even know if the 1 reported by checkstyle is valid.
So this doesn't help your case to make the change later. :)

This raises another question outside of this issue, what happens if we count lines differently then sonar (mis-configuration) and we print a violation on a line which it thinks is outside it's understanding of the size of the file.

> Or would it be acceptable to have this fix first, and do the refactoring later?

I prefer the fix now as I don't see any difference in hurt between your version and what I am asking it to be.

Reply:
ok - stand by ...